husky-fhir/husky-fhir-emed-ch/husky-fhir-emed-ch-epr/src/main/java/org/projecthusky/fhir/emed/ch/epr/resource/ChEmedEprMedicationRequest.java

```diff
@@ -430,4 +429,21 @@ public Author resolveMedicalAuthor() {
     public Instant resolveMedicalAuthorshipTimestamp() {
         return resolveAuthoredOn();
     }
+
+    public boolean hasTreatmentReason() {
+        return this.reasonCode != null && this.getReasonCodeFirstRep().hasText();
+    }
+
+    /**
+     * Gets the treatment reason if available.
+     *
+     * @return the treatment reason or {@code null}.
+     */
+    @Nullable
+    public String getTreatmentReason() {
+        if (this.reasonCode == null || !this.getReasonCodeFirstRep().hasText()) {
```

Comment:
```suggestion
        if (this.reasonCode == null || this.reasonCode.isEmpty() || !this.getReasonCodeFirstRep().hasText()) {
```

Reply:
Good catch, thanks! It should not create the reasonCode on read access.